src/Compilers/VisualBasic/Portable/VisualBasicCompilationOptions.vb

```diff
@@ -272,6 +272,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 metadataImportOptions:=metadataImportOptions,
                 referencesSupersedeLowerVersions:=referencesSupersedeLowerVersions)
 
+            Debug.Assert(globalImports Is Nothing OrElse Not globalImports.Any(Function(x) x Is Nothing))
```

Comment:
>Debug.Assert(globalImports Is Nothing OrElse Not globalImports.Any(Function(x) x Is Nothing)) [](start = 12, length = 93)

It looks like this can come from an external consumer. It feels inappropriate to assert in this case. Either we should validate and throw, or validate later and report a diagnostics, or simply ignore null entries where we consume the value.

Reply:
If I remember correctly, at some point we decided to not throw on construction of invalid options.

---
In reply to: [364892883](https://github.com/dotnet/roslyn/pull/40823#discussion_r364892883) [](ancestors = 364892883)

Reply:
Please let me know which one you would like to see implemented and I can update this pull request.

Reply:
> Please let me know which one you would like to see implemented and I can update this pull request.

Please describe the problem you are trying to address. How it can be observed?

---
In reply to: [364894210](https://github.com/dotnet/roslyn/pull/40823#discussion_r364894210) [](ancestors = 364894210)

Reply:
I tried to use AnalyzerRunner with netcoreapp3.0 from #40824, but it failed with an exception. I'll need to re-run that scenario to get the details again.

Reply:
I updated the original post above to show the stack trace for the error.

Reply:
If someone wants to take over, the steps to see this are:

```
.\Build.cmd -Configuration Release
.\artifacts\bin\AnalyzerRunner\Release\netcoreapp3.0\AnalyzerRunner.exe .\artifacts\bin\AnalyzerRunner\Release\netcoreapp3.0 Roslyn.sln /stats /a CSharpRemoveUnnecessaryCastDiagnosticAnalyzer /a VisualBasicRemoveUnnecessaryCastDiagnosticAnalyzer
```